src/dynamics/models.jl

```diff
@@ -139,6 +139,7 @@ Base.@kwdef struct PrimitiveDryModel{NF<:AbstractFloat, D<:AbstractDevice} <: Pr
     boundary_layer_drag::BoundaryLayerDrag{NF} = LinearDrag(spectral_grid)
     temperature_relaxation::TemperatureRelaxation{NF} = HeldSuarez(spectral_grid)
     static_energy_diffusion::VerticalDiffusion{NF} = StaticEnergyDiffusion(spectral_grid)
+    vertical_advection::VerticalAdvection{NF} = UpwindVerticalAdvection(spectral_grid)
```

Comment:
This is now 5th order upwind for the new default, right? If that's a good choice happy with it, I'll let you judge.

Reply:
Probably `WENOVerticalAdvection(spectral_grid)` is the right choice. `UpwindVerticalAdvection(spectral_grid, order = 3)` might be too diffusive (we don't want to have it as a default), and `UpwindVerticalAdvection(spectral_grid, order = 5)` might be unstable, so WENO should hit the sweet spot.
On the other hand, `WENOVerticalAdvection` might be more computationally expensive, so we could also leave the centered scheme as default but point out different choices in case of instabilities.

Reply:
Making `CenteredVerticalAdvection` default.